### ISSUE TITLE
[v.1.13.0] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#86312]
### 🐛 Describe the bug

We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.13) for the 1.13.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 10/21): work on finalizing the release branch
* Phase 2 (after 10/21): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 10/21/22):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.12 for 1.13 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Fixes to new features introduced in the most recent minor release (e.g. 1.12.0 for 1.13 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @seemethere ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 10/21/22):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.13).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/1.13
    git checkout release/1.13
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.13' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/1.13
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F1.13)

### Versions

1.13.0

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* n/a

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86333

Criteria Category:
* Release branch specific changes (remove prototype feature)

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86314

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86377

Criteria Category:
*  Fix full jit builds during our CircleCI nightly. Fix is required for ios cocoapod release

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86290

Criteria Category:
* CI: Release only change

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86382

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86385

Criteria Category:
* CI Critical fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86484

Criteria Category:
* CI : Release only change, fix RC names

---
@atalman megred

**Comment by EikanWang** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/85056

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86516

Criteria Category:
*  Fix the performance issue that the for-loop before ExternallCall could not be parallelized.

**Comment by soulitzer** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86260

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86672

Criteria Category:
* bug fix (this PR also introduces a feature, so I'm ok with making a new PR directly to release branch that factors out the bug fix if that is better)

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86602

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86689

Criteria Category:
* Documentation improvements

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* N/A 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86693

Criteria Category:
* Release branch specific changes (change functorch documentation version specifier and links to colab notebooks)

**Comment by jeffdaily** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86369

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86725

Criteria Category:
* bug fix

**Comment by jerryzh168** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86631

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86726

Criteria Category:
* critical fix for crashes
See summary of https://github.com/pytorch/pytorch/pull/86631 for context

**Comment by CaoE** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/84410

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86784

Criteria Category:
* Critical fixes for:[crashes](https://github.com/pytorch/pytorch/issues/86790)

**Comment by malfet** [comment]
Linkt to landed master PR:
 - https://github.com/pytorch/pytorch/pull/86303
 
Link to release branch PR:
 - https://github.com/pytorch/pytorch/pull/86844

Criteria Category:
  - Documentation improvements / Fixes to regressions against the most recent minor release

**Comment by malfet** [comment]
Link to landed master PR:
  - https://github.com/pytorch/pytorch/pull/79086

Link to release branch PR:
  - https://github.com/pytorch/pytorch/pull/86851

Criteria Category:
  - Documentation improvements

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86824

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86895

Criteria Category:
* Critical CI Fix

**Comment by XiaobingSuper** [comment]
Link to landed master PR (if applicable):
* n/a

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86900

Criteria Category:
* Documentation improvements for quantization

---

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86815

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86902

Criteria Category:
* Fixes to regressions against the most recent minor release

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86823

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86903

Criteria Category:
* Documentation improvements

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86904

Criteria Category:
* [CI] Critical fix

**Comment by BowenBao** [comment]
Link to landed master PR (if applicable):
* #86168

Link to release branch PR:
* #86921

Criteria Category:
* bug fix

**Comment by BowenBao** [comment]
Link to landed master PR (if applicable):
* #86716

Link to release branch PR:
* #86923

Criteria Category:
* fix regression

**Comment by BowenBao** [comment]
Link to landed master PR (if applicable):
* #85999

Link to release branch PR:
* #86924

Criteria Category:
* bug fix

**Comment by BowenBao** [comment]
Link to landed master PR (if applicable):
* #86843

Link to release branch PR:
* #86925

Criteria Category:
* bug fix

**Comment by smk2007** [comment]
Link to landed master PR (if applicable):


https://github.com/pytorch/pytorch/pull/86557

https://github.com/pytorch/pytorch/pull/86845
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/86803
Criteria Category:


Fix required for enabling the DirectML out-of-tree backend plugin in 1.13.

**Comment by samdow** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86926

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/86934

Criteria Category:
* Silent correctness of functorch

**Comment by malfet** [comment]
Link to the landed master PR:
- https://github.com/pytorch/pytorch/pull/86257

Link to the release branch PR:
  - https://github.com/pytorch/pytorch/pull/86933

Criteria category:
  - Fixes to regressions against the most recent minor release

---
@malfet: merged

**Comment by thiagocrepaldi** [comment]
Link to landed master PR (if applicable): 
https://github.com/pytorch/pytorch/pull/86223

Link to release branch PR: 
https://github.com/pytorch/pytorch/pull/87122

Criteria Category: 
bug fix

**Comment by clee2000** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86979

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87132

Criteria Category:
* Test/CI fixes

**Comment by NivekT** [comment]
Link to the landed master PR:
* https://github.com/pytorch/pytorch/pull/86497
* https://github.com/pytorch/pytorch/pull/87081

Link to the release branch PR:
* https://github.com/pytorch/pytorch/pull/87140

Criteria category:
* Documentation improvements

**Comment by janeyx99** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86985

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87144

Criteria Category:
1. Fixes regression -- the opt_backend had incorrect defaults

-----------

**Comment by malfet** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/86632

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/87201

Criteria Category: 
 - CI fixes (ignores new test shard that was added on trunk)

**Comment by janeyx99** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87135
* https://github.com/pytorch/pytorch/pull/87199

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87261

Criteria Category:
* Fixes regressions (a mps-specific regression in perf + a more silent regression for contraction order)

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87193

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/87307

Criteria Category:
* CI critical fix

**Comment by mikaylagawarecki** [comment]
Link to landed master PR (if applicable):

* https://github.com/pytorch/pytorch/pull/85186

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/87337

Criteria Category:

Documentation improvement

**Comment by cpuhrsch** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/85265

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/87376

Criteria Category
- Documentation improvement

**Comment by cpuhrsch** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/86220

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/87379

Criteria Category
- Documentation improvement

**Comment by cpuhrsch** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/87107

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/87380

Criteria Category
- Documentation improvement / code warning (not exactly sure)

**Comment by zrphercule** [comment]
Link to landed master PR:

https://github.com/pytorch/pytorch/pull/87106
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/87384
Criteria Category

Fixed asserting bugs in Transformer & added unit test
Documentation improvement

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87383

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87386

Criteria Category:
* 5. Documentation improvements

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* n/a

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87392

Criteria Category:
* Release branch specific changes. The functorch docs build is not showing the correct version on the release branch. This PR fixes it.

**Comment by george-qi** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87387

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87406

Criteria Category:
* Documentation improvement

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87382

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87408

Criteria Category:
* CI keep signal green

**Comment by malfet** [comment]
Link to the landed master PR:
 - https://github.com/pytorch/pytorch/pull/87286

Link to the release branch PR:
 - https://github.com/pytorch/pytorch/pull/87427

Criteria Category:
- 2. Critical fixes for crashes (crashes on x86 Mac when bitwise not is used on an empty tensor)

**Comment by malfet** [comment]
Link to landed master PRs:
- https://github.com/pytorch/pytorch/pull/87143
- https://github.com/pytorch/pytorch/pull/86812

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/87443

Criteria Category: Special dispensation (Addressing security concern), needs to be confirmed

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/85953

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87454

Criteria Category:
*  Test/CI fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86959

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87455

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/82379

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87456

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/86745

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87457

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87303

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87463

Criteria Category:
* Critical fix

**Comment by malfet** [comment]
Link to landed master PRs:
 - https://github.com/pytorch/pytorch/pull/86958
 - https://github.com/pytorch/pytorch/pull/86956

Link to release branch PR:
 - https://github.com/pytorch/pytorch/pull/87475

Criteria Category:
  -  Critical fixes for silent correctness

**Comment by atalman** [comment]
Final RC have been generated. We don't Accept cherry-picks anymore. Please don't post any cherry-picks.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87385

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87613

Criteria Category:
* Documentation

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87615

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87612

Criteria Category:
* Critical distributed fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87611

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87618

Criteria Category:
* Critical CUDA/nvrtc fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/87498

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/87628

Criteria Category:
* CI/CD documentation push

**Comment by malfet** [comment]
1.13.0 has been published, closing this one.
